src/modules/Profile.rb

```diff
@@ -547,53 +511,42 @@ def Compat(__current)
     # @param file [String] path to file
     # @return [Boolean]
     def ReadXML(file)
-      tmp = Convert.to_string(SCR.Read(path(".target.string"), file))
-      l = Builtins.splitstring(tmp, "\n")
-      if !tmp.nil? && Ops.get(l, 0, "") == "-----BEGIN PGP MESSAGE-----"
-        out = {}
-        while Ops.get_string(out, "stdout", "") == ""
-          UI.OpenDialog(
-            VBox(
-              Label(
-                _("Encrypted AutoYaST profile. Enter the correct password.")
-              ),
-              Password(Id(:password), ""),
-              PushButton(Id(:ok), Label.OKButton)
-            )
-          )
-          p = ""
-          button = nil
-          begin
-            button = UI.UserInput
-            p = Convert.to_string(UI.QueryWidget(Id(:password), :Value))
-          end until button == :ok
-          UI.CloseDialog
-          command = Builtins.sformat(
-            "gpg2 -d --batch --passphrase \"%1\" %2",
-            p,
-            file
-          )
-          out = Convert.convert(
-            SCR.Execute(path(".target.bash_output"), command, {}),
-            from: "any",
-            to:   "map <string, any>"
-          )
-        end
-        @current = XML.XMLToYCPString(Ops.get_string(out, "stdout", ""))
+      if !GPG.encrypted_symmetric?(file)
+        content = File.read(file)
+      else
         AutoinstConfig.ProfileEncrypted = true
+        label = _("Encrypted AutoYaST profile.")
 
-        # FIXME: rethink and check for sanity of that!
-        # save decrypted profile for modifying pre-scripts
-        if Stage.initial
-          SCR.Write(
-            path(".target.string"),
-            file,
-            Ops.get_string(out, "stdout", "")
-          )
+        begin
+          if AutoinstConfig.ProfilePassword.empty?
```

Comment:
Previously you also check the `nil` case. It is not needed in this case?

Reply:
by default it is empty string and overwrite only by us. So I think not needed.